public_html/wp-content/plugins/camptix/camptix.php

```diff
@@ -5395,6 +5400,198 @@ function shortcode_callback( $atts ) {
 	}
 
+	/**
+	 * Detect condition where tickets page updates might break things.
+	 *
+	 * @param  integer $post_id Post ID to test.
+	 *
+	 * @return boolean          True if updating page can break things.
+	 */
+	public function should_prevent_tickets_page_update( $post_id ) {
+		if ( wp_is_post_revision( $post_id ) ) {
+			return false;
+		}
+
+		if ( 'page' !== get_post_type( $post_id ) ) {
+			return false;
+		}
+
+		// Get the post data as it is, before update.
+		$post_before = get_post( $post_id );
+
+		// Allow save if not tickets page.
+		if ( ! has_shortcode( $post_before->post_content, 'camptix' ) ) {
+			return false;
+		}
+
+		// Allow save until the page has been published.
+		if ( 'publish' !== $post_before->post_status ) {
+			return false;
+		}
+
+		// Allow save if no tickets bought yet.
+		if ( ! wp_count_posts( 'tix_attendee' )->publish ) {
+			return false;
+		}
+
+		return true;
+	}
+
+	/**
+	 * On tickets page update, check if updates are something that can break things after tickets page
+	 * have been published and tickets have been sold. Camptix send ticket emails with links to the
+	 * tickets page, and certain changes to the page do break those links. By not allowing certain updates,
+	 * we try to mitigate the risk of organisers doing odd stuff after ticket sales has opened.
+	 *
+	 * Hooked filter `wp_insert_post_empty_content` only short circuits the save, so we need to handle
+	 * sending meaningful error message ourselves. Otherwise the post wouldn't just update and user would
+	 * get confused with not seeing the applied changes.
+	 *
+	 * @param  boolean $maybe_empty Whether the post should be considered "empty".
+	 * @param  array   $postarr     Array of post data.
+	 *
+	 * @return boolean              Whether the post should be considered "empty". When update should be prevented, send error message.
+	 */
+	public function maybe_prevent_tickets_page_update( $maybe_empty, $postarr ) {
+		if ( ! $this->should_prevent_tickets_page_update( $postarr['ID'] ) ) {
+			return $maybe_empty;
+		}
+
+		// Cannot remove the camptix shortcode anymore.
+		if ( ! has_shortcode( $postarr['post_content'], 'camptix' ) ) {
+			$maybe_empty = true;
+			$message =
+				__( 'You cannot remove the <code>camptix</code> shortcode from the page because tickets have been sold.', 'wordcamporg' ) . ' ' .
+				__( 'Doing that would break the links on ticket emails sent to attendees.', 'wordcamporg' );
+		}
+
+		// Cannot change the visibility of the page anymore.
+		if ( 'publish' !== $postarr['post_status'] || ! empty( $postarr['post_password'] ) ) {
+			$maybe_empty = true;
+			$message =
+				__( 'You cannot unpublish or make the page private because tickets have been sold.', 'wordcamporg' ) . ' ' .
+				__( 'Doing that would break the links on ticket emails sent to attendees.', 'wordcamporg' );
+		}
+
+		// Cannot change the slug anymore.
+		if ( $postarr['post_name'] !== $post_before->post_name ) {
```

Comment:
I think after you refactored in 17b1e53f8cec871f8418864c999ede12f73f241a `$post_before` is no longer in scope.